### PR TITLE
Export Tiles sample

### PR DESCRIPTION
**Pull request by nCastle1**
This is the PR for the implementation of the Export Tiles sample. 

Note: The Android implementation deviates from the pattern of multiple map views that is used for the other platforms. This is a workaround for a defect related to the Xamarin.Android rendering implementation.

### CODE REVIEW

**Comment by nCastle1** [comment]
@macca400 Could you please take a look at this when you get a free moment? Thanks! :)

File: src/WPF/ArcGISRuntime.WPF.Samples/Samples/Layers/ExportTiles/ExportTiles.xaml.cs [hunk 1]
```diff
@@ -0,0 +1,264 @@
...
+        }
+
+        private string GetTilePath()
+        {
+            // Return the platform-specific path for storing the tile cache
+            return Environment.ExpandEnvironmentVariables("%TEMP%\\myTileCache.tpk");
+        }
+
+        /// <summary>
+        /// Starts the export job and registers callbacks to keep aprised of job status
```
Comment (dkemlage) [inline src/WPF/ArcGISRuntime.WPF.Samples/Samples/Layers/ExportTiles/ExportTiles.xaml.cs]: Spelling issue: 'apprised' not 'aprised'.

File: src/WPF/ArcGISRuntime.WPF.Samples/Samples/Layers/ExportTiles/ExportTiles.xaml.cs [hunk 2]
```diff
@@ -0,0 +1,264 @@
...
+            String filePath = GetTilePath();
+
+            // Load the saved tile cache
+            TileCache cache = new TileCache(filePath);
+
+            // Load the cache
+            await cache.LoadAsync();
+
+            // Create a tile layer with the cache
+            ArcGISTiledLayer layer = new ArcGISTiledLayer(cache);
```
Comment (dkemlage) [inline src/WPF/ArcGISRuntime.WPF.Samples/Samples/Layers/ExportTiles/ExportTiles.xaml.cs]: Don't name variables the same as ArcGIS Runtime .NET classes. It should be something like 'myLayer' not 'layer'.

File: src/WPF/ArcGISRuntime.WPF.Samples/Samples/Layers/ExportTiles/ExportTiles.xaml.cs [hunk 3]
```diff
@@ -0,0 +1,264 @@
...
+            // Load the layer
+            await layer.LoadAsync();
+
+            // Create the basemap with the layer
+            Map myMap = new Map(new Basemap(layer));
+
+            // Assign the map to the mapview
+            MyMapView.Map = myMap;
+
+            // Create a new symbol for the extent graphic
```
Comment (dkemlage) [inline src/WPF/ArcGISRuntime.WPF.Samples/Samples/Layers/ExportTiles/ExportTiles.xaml.cs]: Give the user a little more comments as what is really happening here with this graphic. 
This graphic is the red box that appears on the MapView and will be used to define the extent of tiles that are retrieved from the ExportTileCacheJob.

File: src/WPF/ArcGISRuntime.WPF.Samples/Samples/Layers/ExportTiles/ExportTiles.xaml.cs [hunk 4]
```diff
@@ -0,0 +1,264 @@
...
+            MyMapView.ViewpointChanged += MyMapView_ViewpointChanged;
+        }
+
+        private void MyMapView_ViewpointChanged(object sender, EventArgs e)
+        {
+            UpdateMapExtent();
+        }
+
+        /// <summary>
+        /// Function used to keep the overlaid preview area marker in position
```
Comment (dkemlage) [inline src/WPF/ArcGISRuntime.WPF.Samples/Samples/Layers/ExportTiles/ExportTiles.xaml.cs]: Perhaps some more comments here as to what this function does. Something like:
Each time the user changes the extent of the MapView (either by a zoom in/out or a pan) this function fires. When the extent changes, the existing red box (aka. envelope) that is overlaid on the MapView gets updated to be 80% of the current extent. It is the extent of the red box that appears on the MapView that is used to define the extent of tiles that are retrieved from the ExportTileCacheJob.

**Comment by nCastle1** [comment]
Hi @dkemlage , I've made all requested changes. Can you please re-review this?

**Comment by dkemlage** [comment]
@nCastle1 Thanks for implementing all of the changes. 
I tested the UI on each platform - OK (with the exception of native Android 7"KitKat (4.4) XHDPI (Android 4.4 - API 19)). I saw that the Android app works on your test machine, so it could be an Emulator issue on my dev box.
I reviewed all the code - OK. 
The screenshots for each platform - OK. 
The metadata.json TypeLinks - OK.
Everything looks great! Good Job. Please Merge.

**Comment by dkemlage** [comment]
@nCastle1 Now that we have new information on a potentially good fix the native Xamarin Android from @gdestigter, lets make some adjustments to the Xamarin Android version before you do the final Merge.

**Comment by dkemlage** [comment]
@nCastle1 The latest Xamarin Android changes have fixed the issue on my test machine.
Please Merge the code.

**Comment by nCastle1** [comment]
@dkemlage thanks for the review and investigation as we got this working. :)

**Pull request merged**